### PR TITLE
add new package : agrep

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/agrep/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Developed 1989-1991 by Udi Manber, Sun Wu et al. at the University
+    of Arizona."""
+
+    homepage = "http://www.tgries.de/agrep"
+    url      = "https://www.tgries.de/agrep/agrep-3.41.tgz"
+
+    version('3.41', sha256='0508eafaf9725fc67cc955eb6d32ba4f50138443a4fea4275508d2c3f67a234e')
+
+    def install(self, spec, prefix):
+        install('agrep', prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/agrep/package.py]: I would install this to prefix.bin so it shows up in PATH

**Comment by hartzell** [comment]
agrep has a restrictive license (from the [what is AGREP section here](https://www.tgries.de/agrep/)):

```
AGREP belongs to the University of Arizona, which licenses it (see copyright).

It is not public-domain, but free for non-commercial use. And it comes with source!
```

Errr, now that I check, the [GitHub repo](https://github.com/Wikinaut/agrep) says that it's using an ISC license.

What's the relationship between the GitHub project and the homepage/url in this PR?  What license is being used by the bits that this is pulling?

**Comment by darmac** [comment]
Yes, its license is ISC.

> What's the relationship between the GitHub project and the homepage/url in this PR?

=> The github : https://github.com/Wikinaut/agrep give a homepage link at the end of `README.md` as below :
```
Homepage and references
=======================

* http://www.tgries.de/agrep
* http://webglimpse.net
* https://en.wikipedia.org/wiki/Agrep
* [Wu, S., Manber, U.: "Agrep - A Fast Approximate Pattern-Matching Tool", 1992.](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.48.8488)
* [Wu, S., Manber, U.: "Fast Text Searching With Errors", 1991.](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.20.8854)
```

File: var/spack/repos/builtin/packages/agrep/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/agrep/package.py]: I refined it in this patch: https://github.com/spack/spack/pull/15165/commits/b59df9770cace151a8ae459f3523e9113bde91dc

**Pull request merged**